src/hotspot/share/opto/addnode.cpp

```diff
@@ -886,6 +886,22 @@ Node* XorINode::Ideal(PhaseGVN* phase, bool can_reshape) {
       phase->record_for_igvn(this);
     }
   }
+
+  // Propagate xor through constant cmoves. This pattern can occur after expansion of Conv2B nodes.
+  if (in1->Opcode() == Op_CMoveI && in2->is_Con()) {
+    if (in1->in(CMoveNode::IfFalse)->is_Con() && in1->in(CMoveNode::IfTrue)->is_Con() && in1->in(CMoveNode::Condition)->is_Bool()) {
+      int cmp_op = in1->in(CMoveNode::Condition)->in(1)->Opcode();
+
+      int in2_val = phase->type(in2)->is_int()->get_con();
+      int l_val = phase->type(in1->in(CMoveNode::IfFalse))->is_int()->get_con();
+      int r_val = phase->type(in1->in(CMoveNode::IfTrue))->is_int()->get_con();
+
+      if (cmp_op == Op_CmpI || cmp_op == Op_CmpP) {
+        return new CMoveINode(in1->in(CMoveNode::Condition), phase->intcon(l_val ^ in2_val), phase->intcon(r_val ^ in2_val), TypeInt::INT);
+      }
+    }
```

Comment:
An isa_int() check is needed before doing is_int()->get_con(). Something like below:
    ```
  const TypeInt* in2type = phase->type(in2)->isa_int();
      const TypeInt* ltype = phase->type(in1->in(CMoveNode::IfFalse))->isa_int();
      const TypeInt* rtype = phase->type(in1->in(CMoveNode::IfTrue))->isa_int();

      if (in2type && ltype && rtype) {
        int in2_val = in2type->get_con();
        int l_val = ltype->get_con();
        int r_val = rtype->get_con();

        if (cmp_op == Op_CmpI || cmp_op == Op_CmpP) {
          return new CMoveINode(in1->in(CMoveNode::Condition),
                                phase->intcon(l_val ^ in2_val),
                                phase->intcon(r_val ^ in2_val), TypeInt::INT);
        }
      }
```
This should fix the crash seen in vectorapi test.

Reply:
Ah yep, I missed that when writing the function, and it's a simple fix. Thanks for helping diagnose the issue!